addons/addon-base-raas/packages/base-raas-post-deployment/lib/steps/create-service-catalog-portfolio.js

```diff
@@ -295,13 +300,55 @@ class CreateServiceCatalogPortfolio extends Service {
         DisableTemplateValidation: true,
         Type: 'CLOUD_FORMATION_TEMPLATE',
         Description: productToCreate.description || autoCreateDesc,
+        Name: nextProvArtifact,
       },
       ProductId: productId,
     };
     const data = await servicecatalog.createProvisioningArtifact(params).promise();
     return data.ProvisioningArtifactDetail.Id;
   }
 
+  async getNextArtifactVersion(latestVersionInSC) {
+    // For product artifacts in Service Catalog, versions and names are the same thing
+    // If no previous version found, assign 'v2' (this will help in next update cycle)
+    let returnVal = 'v2';
+    // Only finds version strings that match patterns:
+    // "V<n>", "V<n>.0", "V<n>.0.0", "v<n>", "v<n>.0", "v<n>.0.0"
+    const pattern = /^(v|V)(\d+\.)?(\d+\.)?(\*|\d+)$/;
+    let latestVersionNo;
+    if (latestVersionInSC) {
+      const parsedOutput = latestVersionInSC.match(pattern);
+      this.log.info(`${parsedOutput}`);
+      if (parsedOutput && parsedOutput.length > 1) {
+        const longestMatch = parsedOutput.sort(function(a, b) {
+          return b.length - a.length;
+        })[0];
+        latestVersionNo = longestMatch.substring(1);
+        // Increment major version of the string and assign to the new artifact
+        const versionToCompute = [latestVersionNo]
```

Comment:
I may be missing something but if we know the format is V#.#.# couldn't we just do:
`input.substring(1).split('.')[0]` to get the major string?

Or just use the regex grouping?

Reply:
I think we'll need to cover all the mentioned format, since it could be one of these:
1. V#.#.# (The initially decided default version)
2. v# (The version format from here on)
3. Any other format/string that users could have used while manually adding an artifact

Reply:
You're right. That works too.